packages/types/src/api-key/common/api-key.ts

```diff
@@ -1,23 +1,90 @@
 import { BaseFilterable } from "../../dal"
 
+/**
+ * @interface
+ *
+ * An API key's type.
+ */
 export type ApiKeyType = "secret" | "publishable"
+
+/**
+ * The API key details.
+ */
 export interface ApiKeyDTO {
+  /**
+   * The ID of the API key.
+   */
   id: string
+
+  /**
+   * The token of the API key.
+   */
   token: string
+
+  /**
+   * The redacted form of the API key's token. This is useful
+   * when showing portion of the token. For example `sk_...123`.
+   */
   redacted: string
+
+  /**
+   * The title of the API key.
+   */
   title: string
+
+  /**
+   * The type of the API key.
+   */
   type: ApiKeyType
+
+  /**
+   * The date the API key was last used.
+   */
   last_used_at: Date | null
+
+  /**
+   * Who created the API key.
+   */
   created_by: string
+
+  /**
+   * The date the API key was created.
+   */
   created_at: Date
+
+  /**
+   * Who revoked the API key.
```

Comment:
NIT: I think it's nice to specify what exactly this points to (eg. "the ID of the user that revoked the API key")

Reply:
I was thinking of that too, but also felt like it's somewhat limiting. i.e. if someone is using the module on their own, they can use something other than user ID such as email. However, it makes sense to add an example so I'll do that 👍🏻